packages/cardpay-cli/await-bridged.ts

```diff
@@ -0,0 +1,28 @@
+import HDWalletProvider from 'parity-hdwallet-provider';
+import Web3 from 'web3';
+import { HttpProvider, TokenBridgeHomeSide, getConstant, networkIds } from '@cardstack/cardpay-sdk';
+
+export default async function (
+  network: string,
+  mnemonic: string,
+  fromBlock: number,
+  recipient?: string
+): Promise<void> {
+  let web3 = new Web3(
+    new HDWalletProvider({
+      chainId: networkIds[network],
+      mnemonic: {
+        phrase: mnemonic,
+      },
+      providerOrUrl: new HttpProvider(await getConstant('rpcNode', network)),
```

Comment:
This is why you are getting the errors around the subscription not working--HttpProvider doe snot support subscriptions. This HttpProvider I had to do a bunch of patching to get it to work with the HDWalletProvider. We could try doing something similar with a web socket provider--but it will be like a day or 2 of time to get that working I would imagine if its anything like the HttpProvider...

Reply:
I tried it with the WebsocketProvider. It required similar patches to the HttpProvider and then worked for basic operations, but HDWalletProvider itself doesn't support subscriptions, so that path was a dead end. Instead, I added a polling fallback for when the provider doesn't support subscriptions.

Reply:
Hmmm, yeah I’ve seen that some providers make assumptions around downstream providers (it stacks like middleware), which can mess things up. That is very unfortunate to hear. However I don’t thing you should give up on it. The HDWallet provider is just for the CLI. For the web client you’ll be using the wallet connect provider and I think that one will work with subscriptions. It’s unfortunate that the selection of provider by the client can prevent us from using specific SDK methods, but maybe we can use polling as a fall back when subscriptions aren’t  supported